Exiled.API/Features/DamageHandlers/GenericDamageHandler.cs

```diff
@@ -0,0 +1,219 @@
+    using System;
+
+    using Exiled.API.Enums;
+    using Exiled.API.Features.Items;
+
+    using Footprinting;
+
+    using PlayerStatsSystem;
+
+    using Subtitles;
+
+    /// <summary>
+    /// Allows generic damage to player.
+    /// </summary>
+    public class GenericDamageHandler : PlayerStatsSystem.CustomReasonDamageHandler
+    {
+        private const string DamageTextDefault = "You were damaged by Unknown Cause";
+        private string genericDamageText;
+        private string genericEnvironmentDamageText;
+        private Player player;
+        private DamageType damageType;
+        private DamageHandlers.DamageHandlerBase.CassieAnnouncement customCassieAnnouncement;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GenericDamageHandler"/> class.
+        /// Transform input data to custom generic handler.
+        /// </summary>
+        /// <param name="player"> Current player (Target). </param>
+        /// <param name="attacker"> Attacker. </param>
+        /// <param name="damage"> Damage quantity. </param>
+        /// <param name="damageType"> Damage type. </param>
+        /// <param name="cassieAnnouncement"> Custom cassie announcment. </param>
+        /// <param name="damageText"> Text to provide to player death screen. </param>
+        public GenericDamageHandler(Player player, Player attacker, float damage, DamageType damageType, DamageHandlers.DamageHandlerBase.CassieAnnouncement cassieAnnouncement, string damageText = null)
+            : base(DamageTextDefault)
+        {
+            this.player = player;
+            this.damageType = damageType;
+            this.customCassieAnnouncement = cassieAnnouncement;
+            if (this.customCassieAnnouncement is not null)
+            {
+                this.customCassieAnnouncement.Announcement = customCassieAnnouncement.Announcement ?? $"{player.Nickname} killed by {attacker.Nickname} utilizing {damageType}";
+            }
+
+            this.Attacker = attacker.Footprint;
+            this.AllowSelfDamage = true;
+            this.Damage = damage;
+            this.ServerLogsText = $"GenericDamageHandler damage processing";
+            this.genericDamageText = $"You were damaged by {damageType}";
+            this.genericEnvironmentDamageText = $"Environemntal damage of type {damageType}";
+
+            switch (damageType)
+            {
+                case DamageType.Falldown:
+                case DamageType.Hypothermia:
+                case DamageType.Asphyxiation:
+                case DamageType.Poison:
+                case DamageType.Bleeding:
+                case DamageType.Crushed:
+                case DamageType.FemurBreaker:
+                case DamageType.PocketDimension:
+                case DamageType.FriendlyFireDetector:
+                case DamageType.SeveredHands:
+                    Base = new CustomReasonDamageHandler(damageText ?? genericDamageText, damage, cassieAnnouncement.Announcement);
```

Comment:
Because these may cause death, and honestly, I don't know whether to add a specific type here but CustomReason allows you specify death reason, so that's why it's here